security/src/main/java/io/airlift/security/utils/KerberosUtil.java

```diff
@@ -0,0 +1,173 @@
+package io.airlift.security.utils;
+
+import com.google.common.base.Throwables;
+import io.airlift.log.Logger;
+import io.airlift.security.exception.AuthenticationException;
+import org.ietf.jgss.GSSContext;
+import org.ietf.jgss.GSSCredential;
+import org.ietf.jgss.GSSException;
+import org.ietf.jgss.GSSManager;
+import org.ietf.jgss.GSSName;
+import org.ietf.jgss.Oid;
+
+import javax.security.auth.Subject;
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.security.Principal;
+import java.security.PrivilegedAction;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkState;
+
+public class KerberosUtil
```

Comment:
maybe merge KerberosUtil into CloseableGSSContext is even better, since all those util methods are only used by building GSSContext as for now, anyway.